src/Mailchimp/Manager.php

```diff
@@ -0,0 +1,107 @@
+
+    public function __construct(Driver $driver, ContainerInterface $requestBuildersLocator)
+    {
+        $this->driver = $driver;
+        $this->requestBuildersLocator = $requestBuildersLocator;
+    }
+
+    /**
+     * Creates/updates a Mailchimp member
+     */
+    public function editMember(Adherent $adherent, AdherentCommandInterface $message): void
+    {
+        $requestBuilder = $this->requestBuildersLocator
+            ->get(RequestBuilder::class)
+            ->updateFromAdherent($adherent)
+        ;
+
+        $result = $this->driver->editMember(
+            $requestBuilder->buildMemberRequest($message->getEmailAddress())
+        );
+
+        if ($result) {
+            $this->logger->info(sprintf('Mailchimp member "%s" has been edited', $adherent->getUuidAsString()));
+
+            // Active/Inactive member's tags
+            $result = $this->driver->updateMemberTags(
+                $requestBuilder->createMemberTagsRequest($message->getEmailAddress(), $message->getRemovedTags())
+            );
+
+            if ($result) {
+                $this->logger->info(sprintf('Mailchimp member "%s" tags have been updated', $adherent->getUuidAsString()));
+            }
+        }
+    }
+
+    public function getCampaignContent(AdherentMessageInterface $message): string
+    {
+        if (!$message->getExternalId()) {
+            throw new InvalidCampaignIdException(sprintf('Message "%s" does not have a valid campaign id', $message->getUuid()));
+        }
+
+        return $this->driver->getCampaignContent($message->getExternalId());
+    }
+
+    public function editCampaign(AdherentMessageInterface $message): bool
+    {
+        $requestBuilder = $this->requestBuildersLocator->get(CampaignRequestBuilder::class);
+
+        $editCampaignRequest = $requestBuilder->createEditCampaignRequestFromMessage($message);
+
+        if (!$campaignId = $message->getExternalId()) {
+            if (!$campaignId = $this->driver->createCampaign($editCampaignRequest)) {
+                throw new \RuntimeException(sprintf('Campaign for the message "%s" was not created', $message->getUuid()));
+            }
+
+            $message->setExternalId($campaignId);
+
+            return true;
+        }
+
+        return $this->driver->updateCampaign($campaignId, $editCampaignRequest);
+    }
+
+    public function editCampaignContent(AdherentMessageInterface $message): bool
+    {
+        if (!$message->getExternalId()) {
+            throw new InvalidCampaignIdException(sprintf('Message "%s" does not have a valid campaign id', $message->getUuid()->toString()));
+        }
+
+        $requestBuilder = $this->requestBuildersLocator->get(CampaignRequestBuilder::class);
+
+        if (!$this->driver->editCampaignContent($message->getExternalId(), $requestBuilder->createContentRequest($message))) {
+            $this->logger->warning(sprintf('Campaign content of "%s" message was not modified', $message->getUuid()->toString()));
```

Comment:
Same here : `Campaign content of "%s" message has not been modified`

Reply:
done